lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,300 @@
+    },
+    [ChainId.POLYGON_MUMBAI]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.GOERLI]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+        {
+          routingType: RoutingType.DUTCH_LIMIT,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.SEPOLIA]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.CELO]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.CELO_ALFAJORES]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.BNB]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.AVALANCHE]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.BASE_GOERLI]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.BASE]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.BLAST]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+  };
+
+  private static _chainConfigsWithDependencies: { [chainId: number]: ChainConfig };
+  private static _reverseConfigs: { [routeType: string]: number[] };
+
+  /**
+   * Lazy load the full list
+   * Adds dependencies if they don't already exist
+   */
+  static get chainConfigsWithDependencies(): { [chainId: number]: ChainConfig } {
+    if (ChainConfigManager._chainConfigsWithDependencies) {
+      return ChainConfigManager._chainConfigsWithDependencies;
+    }
+    ChainConfigManager._chainConfigsWithDependencies = ChainConfigManager._chainConfigs;
+    for (const dependencyMapping in ChainConfigManager._routeDependencies) {
+      for (const dependency of ChainConfigManager._routeDependencies[dependencyMapping]) {
+        for (const chainId in ChainConfigManager._chainConfigs) {
+          let dependentPresent = false;
+          let dependencyPresent = false;
+          for (const supportedRoutingType of ChainConfigManager._chainConfigs[chainId].routingTypes) {
+            dependentPresent = dependentPresent || supportedRoutingType.routingType == dependencyMapping;
+            dependencyPresent = dependencyPresent || supportedRoutingType.routingType == dependency;
+          }
+          // If we have the dependent but not the dependency, add it
```

Comment:
in what case would the chainConfigs map be missing dependency type configs? Current understanding is that would be a misconfiguration of the configs, i.e. there should be a config for every valid routingType

Reply:
Updated logic to throw error if dependency is missing (fail fast on dev error)